Source/GMCAbilitySystem/Private/Components/GMCAbilityComponent.cpp

```diff
@@ -1203,32 +1392,208 @@ UGMCAbilityEffect* UGMC_AbilitySystemComponent::ApplyAbilityEffect(TSubclassOf<U
 		return nullptr;
 	}
 
-	
+	TGMASBoundQueueOperation<UGMCAbilityEffect, FGMCAbilityEffectData> Operation;
+	if (CreateEffectOperation(Operation, Effect, InitializationData, bOuterActivation, bOuterActivation ? EGMCAbilityEffectQueueType::ServerAuth : EGMCAbilityEffectQueueType::Predicted) == -1)
+	{
+		UE_LOG(LogGMCAbilitySystem, Error, TEXT("[%20s] %s could not create an effect of type %s!"),
+			*GetNetRoleAsString(GetOwnerRole()), *GetOwner()->GetName(), *Effect->GetName())
+		return nullptr;
+	}
+
 	// We are trying to apply an effect from an outside source, so we will need to go trough a different routing to apply it
 	if (bOuterActivation) {
 		if (HasAuthority()) {
-			FGMCOuterApplicationWrapper Wrapper = FGMCOuterApplicationWrapper::Make<FGMCOuterEffectAdd>(Effect, InitializationData);
-			AddPendingEffectApplications(Wrapper);
+			QueuedEffectOperations.QueuePreparedOperation(Operation, false);
+			ClientQueueEffectOperation(Operation);
```

Comment:
I do wonder if we could instead have another queue on the server, so we don't spam RPCs in case of multiple effects. Minor optimization perhaps but we could have an array of operations and send them as one RPC instead.